api/gatewayserver.proto

```diff
@@ -96,6 +97,17 @@ service NsGs {
   rpc ScheduleDownlink(DownlinkMessage) returns (ScheduleDownlinkResponse);
 }
 
+message BatchGetGatewayConnectionStatsRequest {
+  repeated GatewayIdentifiers gateway_ids = 1 [(validate.rules).repeated.min_items = 1, (validate.rules).repeated.max_items = 100];
+  // The names of the gateway stats fields that should be returned.
+  // This mask will be applied on each entry returned.
+  google.protobuf.FieldMask field_mask = 2;
+}
+
+message BatchGetGatewayConnectionStatsResponse {
+  map<string,GatewayConnectionStats> entries = 1;
```

Comment:
Actually apologies for the confusion but UID isn't necessary here; it can just be the gateway ID. Please also document this in godoc.